rtCommon/bidsRun.py

```diff
@@ -114,31 +145,20 @@ def appendIncremental(self, incremental: BidsIncremental,
 
                 canAppend, metadataErrorMsg = metadataAppendCompatible(
                     incremental.imageMetadata,
-                    self.incrementals[-1].imageMetadata)
+                    self._imageMetadata)
 
                 if not canAppend:
                     errorMsg = ("Incremental's metadata not compatible "
                                 f" with this run's images ({metadataErrorMsg})")
                     raise MetadataMismatchError(errorMsg)
 
-        # Slice up the incremental into smaller incrementals if it has multiple
-        # images in its image volume
+        # Slice up the incremental into smaller component images if it has
+        # multiple images in its image volume
         imagesInVolume = incremental.imageDimensions[3]
-        if imagesInVolume == 1:
-            self.incrementals.append(incremental)
-        else:
-            # Split up the incremental into single-image volumes
-            image = incremental.image
-            imageData = getNiftiData(image)
-            affine = image.affine
-            header = image.header
-            metadata = incremental.imageMetadata
-
-            for imageIdx in range(imagesInVolume):
-                newData = imageData[..., imageIdx]
-                newImage = incremental.image.__class__(newData, affine, header)
-                newIncremental = BidsIncremental(newImage, metadata)
-                self.incrementals.append(newIncremental)
+        imageData = getNiftiData(incremental.image)
+        for imageIdx in range(imagesInVolume):
+            newData = imageData[..., imageIdx]
+            self._dataArrays.append(newData)
```

Comment:
Can we store the data in an in-memory niftiImage instead of a dataArray[]? Or are you doing it this way to handle both nifti1 and nifti2 images?